### DISCUSSION TITLE
Add option to logAndNotify on Microsoft Naming Convention errors

### DISCUSSION THREAD

**Discussion by Kibubu** [abraunegg/onedrive#2147] [Ideas]
While I had to tweak my sync_list to reduce notifications for short lived files I was quite surprised not to see any notifications concerning skipped files due to MS's Naming Convention errors.

**I would propose to add a command line parameter enabling notifications for said skipped files.**

I took a look at the code and guess this is the part that needs to be adjusts

https://github.com/abraunegg/onedrive/blob/738be2d150efe6cf4cb1ad9bd9b8ced15ac4eaad/src/sync.d#L3542-L3545

to look more like this one

https://github.com/abraunegg/onedrive/blob/738be2d150efe6cf4cb1ad9bd9b8ced15ac4eaad/src/main.d#L1340-L1350

I would like to create a PR but wanted to ask if there is any interest or even maybe a better solution first.

A better solution would of course be a subscription system, similar to the sync_list with the possibility to enable/disable specific warnings, skips, errors. But this would require more work.

**Comment by abraunegg** [comment]
@Kibubu 
Thanks for the suggestion.

It is rather trivial to change those lines to be log and notify by default - will look at this in the next few days

**Comment by abraunegg** [comment]
@Kibubu 
This is resolved with PR https://github.com/abraunegg/onedrive/pull/2153

* Update skip notification handling for the following scenarios:
  * Invalid Name (Microsoft Naming Convention)
  * Invalid Name (Contains an invalid whitespace item)
  * Invalid Name (Contains HTML ASCII Code)
  * Invalid Item (Invalid symbolic link)

When these items are triggered, send a notification to the GUI if it has been configured

**Reply by Kibubu** [reply]
Wonderful thank you so much.
